bodo/pandas/groupby.py

```diff
@@ -343,47 +533,89 @@ def _get_aggfunc_str(func):
 
 
 def _get_agg_output_type(func: str, pa_type: pa.DataType, col_name: str) -> pa.DataType:
-    """Gets the output type of an aggregation or raise either TypeError or BodoLibNotImplementedException
-    for unsupported pa_type/func combinations. Should closely match
-    https://github.com/bodo-ai/Bodo/blob/d1133e257662348cc7b9ef52cf445633036133d2/bodo/libs/groupby/_groupby_common.cpp#L562
+    """Cast the input type to the correct output type depending on func or raise if
+    the specific combination of func + input type is not supported.
+
+    Args:
+        func (str): The function to apply.
+        pa_type (pa.DataType): The input type of the function.
+        col_name (str): The name of the column in the input.
+
+    Raises:
+        BodoLibNotImplementedException: If the operation is not supported in Bodo
+            but is supported in Pandas.
+        TypeError: If the operation is not supported in Bodo or Pandas (due to gaps
+            in Pandas' handling of Arrow Types)
+
+    Returns:
+        pa.DataType: The output type from applying func to col_name.
     """
-    new_type = pa_type
-    if func == "sum":
+    new_type = None
+    fallback = False
+
+    # TODO: Enable more fallbacks where the operation is supported in Pandas and not in Bodo
+    if func in ("sum"):
```

Comment:
Subtle tuple vs. scalar value bug:
```suggestion
    if func in ("sum",):
```